### PR TITLE
Add milestones document

### PR DESCRIPTION
**Pull request by pmorie**
Add milestones that were discussed at the Dec 15, 2016 SIG meeting.

### CODE REVIEW

File: docs/v1/milestones.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+## Legos-1
+
+Tentative target date: Feb 1, 2017
+
+1.  `servicecatalog/v1alpha1` API complete and functional:
+  1.  `Broker`, `ServiceClass`, `Instance`, `Binding`
+  2.  Integrators should be able to program against this REST API
+2.  API server serves `servicecatalog/v1alpha1`
+3.  Golang client for API in (1) is auto-generated
```
Comment (duglin) [inline docs/v1/milestones.md]: Should we include a statement about being able to to CRUD ops via kubectl ?

Comment (krancour) [inline docs/v1/milestones.md]: Although not explicitly stated, I think that falls under MVP-1 item 3:

> Native-k8s-style CLI experience

i.e. I don't think @pmorie intended that as a goal for the Legos-1 milestone.

Comment (duglin) [inline docs/v1/milestones.md]: Just for my understanding... at the end of legos-1, what would prevent someone from using kubectl to CRUD our resources?

Comment (pmorie) [inline docs/v1/milestones.md]: @duglin 

> at the end of legos-1, what would prevent someone from using kubectl to CRUD our resources?

Very good question.  You will be able to use kubectl for _raw_ CRUD, but you won't get a nicely formatted list / describe.  More like the TPR experience.

Comment (duglin) [inline docs/v1/milestones.md]: ok cool - that's what I expected and was hoping for.

File: docs/v1/milestones.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  1.  `Broker`, `ServiceClass`, `Instance`, `Binding`
+  2.  Integrators should be able to program against this REST API
+2.  Golang client for API in (1) is auto-generated
+3.  Native-k8s-style CLI experience
+  1.  Formatted list
+  2.  Formatted describe
+4.  Integration with Open Service Broker API:
+  1.  Add broker
+  2.  Provision service instance
+  3.  Bind to service instance
```
Comment (duglin) [inline docs/v1/milestones.md]: Can we add something like?
```
3. Bind to a service instance.  Credentials returned from a SB will be stored as either secrets
or configMaps. Pod authors will be expected to manually reference the creds from their Pod
specs to use them.
```
Assuming this is what people were thinking.

File: docs/v1/milestones.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+5.  Binding manifests as a Secret in the the k8s core
+
+High-level architectural requirements:
+
+1.  We are moved off TPR and fully onto `servicecatalog/v1alpha1` API
+2.  Code generators for API are in place:
+  1.  Deep copy
+  2.  Defaults
+  3.  Conversions
+  4.  YAML parser
```
Comment (duglin) [inline docs/v1/milestones.md]: Can you refresh my memory - what is the YAML parser for?

Comment (arschles) [inline docs/v1/milestones.md]: iirc it's the hard-coded (more performant than the reflection based) parser for incoming resource manifests

File: docs/v1/milestones.md (hunk 2 above)
Comment (arschles) [inline docs/v1/milestones.md]: @duglin that looks pretty good. one point of clarification - iirc we were gonna store all returned credentials in a single secret for MVP-1. is that your understanding?

Comment (duglin) [inline docs/v1/milestones.md]: I can't recall where we agreed they would go, but a single secret is fine with me.

File: docs/v1/milestones.md (hunk 3 above)
Comment (duglin) [inline docs/v1/milestones.md]: Interesting that they don't have one for json too

**Comment by duglin** [comment]
SGTM just waiting for a minor tweak before I LGTM

File: docs/v1/milestones.md (hunk 3 above)
Comment (MHBauer) [inline docs/v1/milestones.md]: json is yaml.

File: docs/v1/milestones.md (hunk 2 above)
Comment (pmorie) [inline docs/v1/milestones.md]: I am elaborating on this in (5) in the existing list

**Comment by pmorie** [comment]
> We should however explicitly address, here or in a follow-up PR, when we'll support asynchronous provision/deprovision operations.

I agree, and would prefer to do that in a follow-up.

**Comment by pmorie** [comment]
Addressed nits, @duglin PTAL

**Comment by duglin** [comment]
LGTM

**Pull request merged**